src/main/resources/static/common/js/demoAdmin.js

```diff
@@ -1544,20 +1544,35 @@ mod.controller('optionsForm', function($scope, $rootScope, $http, $filter) {
                     }
                 ];
             } else if(options.mqType === "KAFKA") {
+                var kafkaServer = data.kafkaConfig.bootstrapServers;
+                if (kafkaServer.startsWith("localhost:")) {
+                    kafkaServer = data.kafkaConfig.bootstrapServers.replace("localhost", location.hostname);
+                }
+                if (kafkaServer.startsWith("127.0.0.1:")) {
+                    kafkaServer = data.kafkaConfig.bootstrapServers.replace("127.0.0.1", location.hostname);
+                }
+                // Special handling for docker started with original docker-compose.yml file
+                if (kafkaServer === "kafka:9092") {
```

Comment:
@awang-parasoft I have some question about this special handling before we merge.
Is this prefix "kafka:xxx" using the service name we specified in the docker-compose.yaml file? 
Is it possible that user can start their own Kafka container and use same prefix to access the server?

If that's the case, it's possible that they expose the port differently (not have to be 9093). And we should not do special handling like this. Another solution is needed.

If not, go ahead and merge this.

Reply:
> @awang-parasoft I have some question about this special handling before we merge. Is this prefix "kafka:xxx" using the service name we specified in the docker-compose.yaml file? Is it possible that user can start their own Kafka container and use same prefix to access the server?
> 
> If that's the case, it's possible that they expose the port differently (not have to be 9093). And we should not do special handling like this. Another solution is needed. It may be ok to display the internal port since they should realized that it's inside the docker and should go to check the external port.
> 
> If not, go ahead and merge this.

* Yes, the prefix `kafka:xxx` is using the service name in the *docker-compose* file.
* Yes. I have tried to create a network, start a Kafka container with the `kafka` and connect to the network, then start PDA container with `kafka:9092` config and connect to the network, it's the same behaviour as using the *docker-compose* file.

Users can expose a different port when starting their own Kafka server, we only handle the case when they start PDA with the provided unmodified *docker-compose* file. I also think it's OK to display the internal port, as the **TEST CONNECTION** button currently tests connection to Kafka server with the internal port.

Reply:
Ok, then my understanding is that it's better to remove this special handling and just display the internal port. 
@rliu-parasoft Can you please make changes for it? Please talk to Alfie if there's confusion.